pkg/pointerdb/pointerdb.go

```diff
@@ -153,6 +153,7 @@ func (s *Server) Get(ctx context.Context, req *pb.GetRequest) (resp *pb.GetRespo
 		node, err := s.cache.Get(ctx, piece.NodeId)
 		if err != nil {
 			s.logger.Error("Error getting node from cache")
+			continue
```

Comment:
Didn't receiver assume that the piece num and node index match in the ndoes?

Reply:
You are right. Pushed another commit.

Reply:
yeah